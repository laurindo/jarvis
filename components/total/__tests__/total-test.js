import React from "react";
import {render} from "@testing-library/react-native";
import HeaderTotal from "../header-total";
import {create} from "react-test-renderer";

describe('<HeaderTotal/>', () => {
	it('match snapshot', () => {
		const tree = create(<HeaderTotal value={2000}/>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('render HeaderTotal with valid number', () => {
		const {getByText} = render(<HeaderTotal value={2000}/>);
		expect(getByText('R$ 2000')).toBeTruthy();
	});

	it('render HeaderTotal with invalid number', () => {
		const {getByText} = render(<HeaderTotal value={"298,87"}/>);
		expect(getByText('R$ 0')).toBeTruthy();
	});

	it('render HeaderTotal with invalid number', () => {
		const {getByText} = render(<HeaderTotal value={"abv298,87"}/>);
		expect(getByText('R$ 0')).toBeTruthy();
	});
});
