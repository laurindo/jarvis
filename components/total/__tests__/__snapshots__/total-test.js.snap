// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderTotal/> match snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingLeft": 16,
      "paddingRight": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans-Bold",
          "fontSize": 32,
          "marginBottom": 24,
          "marginTop": 24,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <Text>
      R$
       
      2000
    </Text>
  </Text>
</View>
`;
